homeassistant/components/hyperion/config_flow.py

```diff
@@ -0,0 +1,379 @@
+        # wait on the response (which includes the user needing to visit the Hyperion
+        # UI to approve the request for a new token).
+        self._request_token_task = self.hass.async_create_task(
+            self._request_token_task_func(self._auth_id)
+        )
+        return self.async_external_step(
+            step_id="create_token_external", url=self._get_hyperion_url()
+        )
+
+    async def async_step_create_token_external(
+        self, auth_resp: Optional[ConfigType] = None
+    ) -> Dict[str, Any]:
+        """Handle completion of the request for a new token."""
+        if client.ResponseOK(auth_resp):
+            token = auth_resp.get(const.KEY_INFO, {}).get(const.KEY_TOKEN)
+            if token:
+                self._data[CONF_TOKEN] = token
+                return self.async_external_step_done(
+                    next_step_id="create_token_success"
+                )
+        return self.async_external_step_done(next_step_id="create_token_fail")
+
+    async def async_step_create_token_success(
+        self, _: Optional[ConfigType] = None
+    ) -> Dict[str, Any]:
+        """Create an entry after successful token creation."""
+        # Clean-up the request task.
+        await self._cancel_request_token_task()
+
+        # Test the token.
+        async with await self._create_client() as hyperion_client:
+            if not hyperion_client:
+                return self._show_auth_form(
+                    {CONF_BASE: "auth_new_token_not_work_error"}
+                )
+        return await self.async_step_confirm()
+
+    async def async_step_create_token_fail(
+        self, _: Optional[ConfigType] = None
+    ) -> Dict[str, Any]:
+        """Show an error on the auth form."""
+        # Clean-up the request task.
+        await self._cancel_request_token_task()
+        return self._show_auth_form({CONF_BASE: "auth_new_token_not_granted_error"})
+
+    async def async_step_confirm(
+        self, user_input: Optional[ConfigType] = None
+    ) -> Dict[str, Any]:
+        """Get final confirmation before entry creation."""
+        if user_input is None and self._require_confirm:
+            return await self._show_confirm_form()
+
+        async with await self._create_client() as hyperion_client:
+            if not hyperion_client:
+                return self._show_setup_form({CONF_BASE: "connection_error"})
+            hyperion_id = await hyperion_client.async_id()
+
+        if not hyperion_id:
+            return self.async_abort(reason="no_id")
+
+        await self.async_set_unique_id(hyperion_id)
+        self._abort_if_unique_id_configured()
+
+        # pylint: disable=no-member  # https://github.com/PyCQA/pylint/issues/3167
+        return self.async_create_entry(
+            title=f"{self._data[CONF_HOST]}:{self._data[CONF_PORT]}", data=self._data
+        )
+
+    def _show_setup_form(self, errors: Optional[Dict] = None) -> Dict[str, Any]:
```

Comment:
There's rarely a reason to break out the form calls into methods since they are not reusable by other steps. I suggest we inline these in the step methods where they belong.